src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1153,72 @@ const isMobileFieldUpdateAllowed = (
     },
   )
 }
+
+export const createTwilioCredentials = (
+  originalForm: IPopulatedForm,
+  twilioCredentials: TwilioCredentials,
+  formId: string,
+): ResultAsync<SecretsManager.Types.CreateSecretRequest, ApplicationError> => {
+  const msgSrvcName = `formsg/${process.env.FORMSG_SDK_MODE}/form/${formId}/twilio`
+
+  // TO DO: Add error handling to undo MongoDB
+  void updateForm(originalForm, { msgSrvcName })
```

Comment:
should the update be on the model layer? i.e. `FormModel.updateByMsgSrvcName(formId, msgSrvcName)`.
If not, it can even be a separate service function that ONLY updates Twilio, something like `admin-form.service#updateFormSettings`

`updateForm` is a deprecated function (that is not being denoted as such 😓 ), and we should not use it

Reply:
I think you are right, the update should be done on the model layer :D